R/transform-code.R

```diff
@@ -15,40 +15,78 @@ transform_code <- function(path, fun, verbose = FALSE, ...) {
       fun = partial(transform_rmd, transformer_fun = fun), ...,
       verbose = verbose
     )
+  } else if (is_rnw_file(path)) {
+    enc::transform_lines_enc(path,
+      fun = partial(transform_rnw, transformer_fun = fun), ...,
+      verbose = verbose
+    )
   } else {
-    stop(path, " is not an R or Rmd file")
+    stop(path, " is not an R, Rmd or Rnw file")
   }
 }
 
-#' Transform Rmd contents
+#' Transform mixed contents
 #'
 #' Applies the supplied transformer function to code chunks identified within
-#' an Rmd file and recombines the resulting (styled) code chunks with the text
+#' an Rmd or Rnw file and recombines the resulting (styled) code chunks with the text
 #' chunks.
 #'
-#' @param lines A character vector of lines from an Rmd file
+#' @param lines A character vector of lines from an Rmd or Rnw file
 #' @param transformer_fun A styler transformer function
+#' @param filetype A string indicating the filetype (Rmd or Rnw)
 #' @importFrom purrr flatten_chr
 #' @keywords internal
-transform_rmd <- function(lines, transformer_fun) {
-  chunks <- separate_chunks(lines)
+transform_mixed <- function(lines, transformer_fun, filetype = "Rmd") {
+  chunks <- separate_chunks(lines, filetype)
   chunks$r_chunks <- map(chunks$r_chunks, transformer_fun)
 
   map2(chunks$text_chunks, c(chunks$r_chunks, list(character(0))), c) %>%
     flatten_chr()
 }
 
+#' Transform Rmd contents
+#'
+#' Applies the supplied transformer function to code chunks identified within
+#' an Rmd file and recombines the resulting (styled) code chunks with the text
+#' chunks.
+#'
+#' @param lines A character vector of lines from an Rmd file
+#' @param transformer_fun A styler transformer function
+#' @importFrom purrr flatten_chr
+#' @keywords internal
+transform_rmd <- function(lines, transformer_fun) {
+  transform_mixed(lines, transformer_fun, filetype = "Rmd")
```

Comment:
Yeah I think `purrr:partial()` is more elegant. I've now reworked this since I no longer feel the explicit `transform_rmd()` and `transform_rnw()` functions are necessary.

Reply:
👍